src/lib.rs

```diff
@@ -443,6 +443,59 @@ impl Config {
         if !self.defined("CMAKE_TOOLCHAIN_FILE") {
             if let Some(s) = self.getenv_target_os("CMAKE_TOOLCHAIN_FILE") {
                 self.define("CMAKE_TOOLCHAIN_FILE", s);
+            } else {
+                if target.contains("redox") {
+                    if !self.defined("CMAKE_SYSTEM_NAME") {
+                        self.define("CMAKE_SYSTEM_NAME", "Generic");
+                    }
+                } else if target.contains("solaris") {
+                    if !self.defined("CMAKE_SYSTEM_NAME") {
+                        self.define("CMAKE_SYSTEM_NAME", "SunOS");
```

Comment:
We can also address https://github.com/rust-lang/cmake-rs/issues/151 by removing this if it's acceptable.

Reply:
Yes, please!